src/compiler/preprocess/index.ts

```diff
@@ -156,11 +160,34 @@ function decoded_sourcemap_from_generator(generator: any) {
 	return map;
 }
 
+/**
+ * Heuristic used to find index of component source inside source map sources.
+ */
+function guess_source_index(
+	file_basename: string,
+	decoded_map: DecodedSourceMap
+): number {
+	if (!file_basename) {
+		return decoded_map.sources.findIndex(source => !source);
+	}
+	// different tools produce different sources
+	// (file name, relative path, absolute path)
```

Comment:
can we add a test for these different cases?

Reply:
Added `guess-source`. I'm not sure if this is the right approach though... While it may workaround some cases, maybe it would be better to require correct maps from preprocessors?

Reply:
I do think it would be a good idea to have preprocessors producing more consistent sourcemaps either way even if we didn't require them here. Requiring them sounds like a reasonable idea though. Do you know which ones are producing these different paths and what the preferred version is? It could be helpful to file an issue against https://github.com/sveltejs/svelte-preprocess with these details

Reply:
I had issue with postcss transformer. The first thing that comes to mind - it should pass "to" parameter to postcss (equal to "from"). Will check and report later.